fv3config/fv3run/_kubernetes.py

```diff
@@ -0,0 +1,266 @@
+            memory_gb=3.6, memory_gb_limit=None, cpu_count=1, gcp_secret=None,
+            image_pull_policy='IfNotPresent'):
+        """Container for kubernetes-specific job configuration.
+
+        Args:
+            jobname (str, optional): name to use for the kubernetes job, defaults to a
+                random uuid.uuid4().hex
+            google_cloud_project (str, optional): value for GOOGLE_CLOUD_PROJECT
+                environment variable when running the kubernetes job, which is used by
+                gcsfs for some operations. By default the environment variable is unset.
+            memory_gb (float, optional): gigabytes of memory required for the kubernetes
+                worker, defaults to 3.6GB
+            memory_gb_limit (float, optional): maximum memory allowed for the kubernetes
+                worker, defaults to the value set by memory_gb
+            cpu_count (int, optional): number of CPUs to use on the kubernetes worker
+            gcp_secret (str, optional): name of kubernetes secret to mount containing a
+                file ``key.json`` to use as the google cloud storage key.
+            image_pull_policy (str, optional): pull policy passed on to the
+                kubernetes job. if set to "Always", will always pull the latest image.
+                When "IfNotPresent", will only pull if no image has already been pulled.
+                Defaults to "IfNotPresent".
+        """
+        if jobname is None:
+            self.jobname = uuid.uuid4().hex
+        else:
+            self.jobname = jobname
+        if google_cloud_project is None:
+            try:
+                self.google_cloud_project = filesystem._get_gcloud_project()
+            except ImportError:
+                self.google_cloud_project = None
+        else:
+            self.google_cloud_project = google_cloud_project
+        self.memory_gb = memory_gb
+        if memory_gb_limit is None:
+            self.memory_gb_limit = memory_gb
+        else:
+            self.memory_gb_limit = memory_gb_limit
+        self.cpu_count = cpu_count
+        self.gcp_secret = gcp_secret
+        self.image_pull_policy = image_pull_policy
+
+    @property
+    def resource_requirements(self):
+        return kube.client.V1ResourceRequirements(
+            limits={
+                'memory': f'{self.memory_gb_limit:.1f}G',
+            },
+            requests={
+                'memory': f'{self.memory_gb:.1f}G',
+                'cpu': f'{self.cpu_count:.1f}'
+            },
+        )
+
+    @property
+    def _secret_volume(self):
+        if self.gcp_secret is not None:
+            return kube.client.V1Volume(
+                name='gcp-key-secret',
+                secret=kube.client.V1SecretVolumeSource(
+                    secret_name=self.gcp_secret
+                )
+            )
+        else:
+            return None
+    
+    @property
+    def volume_list(self):
```

Comment:
Nit: maybe rename to `volumes`. I am not a huge fan of naming things with the type in them unnecessarily. Is it really important that it's a "list" versus another type of iterable?

Reply:
Having both `varname` and `varnames` is a specific example in _Clean Code_ of something not to do, since it's hard to differentiate the two at a glance, and easy to accidentally typo one into the other. As a result I often append `_list` to an iterable of things, that does not necessarily need to be of type `list`.

Here we don't have both names, and the name used by kubernetes is also `volumes` and `volume_mounts`, so I've renamed it as you suggested.

Reply:
Yah maybe it's having two names which is the problem. I can't think of a case when a good software unit would have both `volume` and `volumes` attributes since that would indeed be confusing.

I don't remember that section in Clean Code. Is there something wrong with this?
```
for name in names:
   pass
```
That seems like good idiomatic python code to me.

Reply:
Apparently it wasn't in Clean Code, I must have read it somewhere else. I do find though that it's easier to read at a glance:
```
for name in name_list:
    pass
```
I also have been burned at least twice by using `for name in names` where I accidentally wrote `names` (or the equivalent for the name I had) within the for loop.

I agree that when they don't appear together, the simple plural noun is probably best.